backend/pkg/modules/internal/analytics_core/init.go

```diff
@@ -40,17 +44,19 @@ const (
 type TopicType string
 
 type AnalyticsCore struct {
-	httpHandler         http.Handler
-	msgBroker           *pubsub.Handler
-	accessor            core.BackendAccessor
-	info                *core.ModuleInfo
-	numWorkers          int
-	proccFuncRegistered map[TopicType][]AnalyticsModuleProccFunc
-	topics              []TopicType
+	httpHandler          http.Handler
+	msgBroker            *pubsub.Handler
+	accessor             core.BackendAccessor
+	info                 *core.ModuleInfo
+	numWorkers           int
+	proccFuncRegistered  map[TopicType][]AnalyticsModuleProccFunc
+	dataFramesRegistered map[AnalyticsModuleProccFunc]*ProcFuncDataFrames // Each registered function have a correponding Dataframes
+	topics               []TopicType
 }
 
+
 type ProcFuncDataFrames struct {
-	dataFrames map[int]*interface{}
+	dataFrames map[int]dataframe.DataFrame
```

Comment:
The idea is that each AnalyticsModuleProccFunc may have more than one data frame. So we shall have something like map[string] map[int] dataframe.DataFrame...

Or better add a structure layer to encapsulate the internal map.